### PR TITLE
Add default value on default local

### LINKED ISSUE #11149: Unable to use a select attribute with a value without a label
**Sylius version affected**: 1.6.5

**Description**  
This error `Notice: Undefined index: fr_FR` occurs when you want to add a select attribute with an empty value onto the product.

**Steps to reproduce**  
1. Add a new attribute with type select.
1. On value list, keep one without value and save.
1. Now, edit a product and attempt to add the attribute added in the previous step.

**Possible Solution**  

1. Return empty value if the translation key does not exist.
1. Use a fallback language to return a value.

### PR DESCRIPTION
**Pull request by macintoshplus**
Add default value if no translation exist on default local

| Q               | A
| --------------- | -----
| Branch?         | 1.6
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #11149
| License         | MIT

### CODE REVIEW

**Comment by macintoshplus** [comment]
Hello @lchrusciel,

I have tested on my Sylius project before send this PR. I dont know how you want test a Symfony FormType : Behat, PHPSpec, PHP Unit ?

**Comment by lchrusciel** [comment]
We already have a few form type tests, like this one: https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/AddressingBundle/Tests/Form/Type/ZoneChoiceTypeTest.php

Could you try to reproduce and perhaps cover this issue, so we will not have a regression in the future?

**Comment by macintoshplus** [comment]
@lchrusciel I hav added the test.

File: config/bundles.php [hunk 1]
```diff
@@ -1,16 +1,5 @@
...
-/*
- * This file is part of the Sylius package.
- *
- * (c) Paweł Jędrzejewski
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-declare(strict_types=1);
```
Comment (lchrusciel) [inline config/bundles.php]: Can you please, bring this content back?

Comment (macintoshplus) [reply config/bundles.php]: Sorry, I have restored.

**Comment by macintoshplus** [comment]
> I would need to ask you to rebase this PR to the newest changes on branch 1.6

I have rebase on branch 1.6

**Comment by lchrusciel** [comment]
It's something wrong with the branch. We need to wait for #11179 to merge this PR

**Comment by lchrusciel** [comment]
Can you rebase again? Build should be green now

**Comment by macintoshplus** [comment]
@lchrusciel Sorry, after many rebase, the tests does not pass

**Comment by macintoshplus** [comment]
Hi @lchrusciel,

I have rebase to Sylius 1.7. But tests fail. Why ?

**Comment by macintoshplus** [comment]
Related #11150

File: src/Sylius/Bundle/AttributeBundle/Form/Type/AttributeType/SelectAttributeType.php [hunk 2]
```diff
@@ -95,7 +95,7 @@ public function configureOptions(OptionsResolver $resolver): void
                             continue;
                         }
 
-                        $choices[$key] = $choice[$this->defaultLocaleCode];
+                        $choices[$key] = $choice[$this->defaultLocaleCode] ?? '';
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AttributeBundle/Form/Type/AttributeType/SelectAttributeType.php:98]: As I came here once again. I wanted to ask one more thing. If I understood correctly, the choice will be just empty in this case. What will happen with more than one undefined choice? Wouldn't it be better to just skip values like this? What kind of behavior would you expect to have here?

Comment (macintoshplus) [reply src/Sylius/Bundle/AttributeBundle/Form/Type/AttributeType/SelectAttributeType.php:98]: In first time, I want only solve the problem.
It's true, the value need to be skipped. But why, I will be change my fix and the test, if check never pass ?

Comment (lchrusciel) [reply src/Sylius/Bundle/AttributeBundle/Form/Type/AttributeType/SelectAttributeType.php:98]: I would say, that removing the empty string from https://github.com/Sylius/Sylius/pull/11150/files#diff-64baa708ca6b415da2aa24111f3655dcR30 would be enough.

Comment (macintoshplus) [reply src/Sylius/Bundle/AttributeBundle/Form/Type/AttributeType/SelectAttributeType.php]: @lchrusciel I have change the behavior. It's OK ?

File: src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        $this->translationProvider = $this->prophesize(TranslationLocaleProviderInterface::class);
+        $this->translationProvider->getDefaultLocaleCode()->willReturn('en_GB');
+
+        parent::setUp();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getExtensions()
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php:62]:
```suggestion
    protected function getExtensions(): iterable
```

File: src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        $this->translationProvider = $this->prophesize(TranslationLocaleProviderInterface::class);
+        $this->translationProvider->getDefaultLocaleCode()->willReturn('en_GB');
+
+        parent::setUp();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php:61]:
```suggestion
```

File: src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        $form = $this->factory->create(\Sylius\Bundle\AttributeBundle\Form\Type\AttributeType\SelectAttributeType::class,
+            null, $formConfiguration);
+        $view = $form->createView();
+
+        Assert::assertSame($expectedLabels, array_map(function (ChoiceView $choiceView): string {
+            return $choiceView->label;
+        }, $view->vars['choices']));
+    }
+
+    protected function setUp()
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php:51]:
```suggestion
    protected function setUp(): void
```

File: src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            'configuration' => [
+                'multiple' => false,
+                'min' => null,
+                'max' => null,
+                'choices' => ['val1' => ['en_GB' => 'value 1'], 'val2' => ['fr_FR' => 'valeur 2']],
+            ],
+        ]);
+    }
+
+    private function assertChoicesLabels(array $expectedLabels, array $formConfiguration = []): void
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php:40]: The private method should be placed after protected

File: src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+use Symfony\Component\Form\ChoiceList\View\ChoiceView;
+use Symfony\Component\Form\PreloadedExtension;
+use Symfony\Component\Form\Test\TypeTestCase;
+
+class SelectAttributeTypeTest extends TypeTestCase
+{
+    /** @var ProphecyInterface|TranslationLocaleProviderInterface */
+    private $translationProvider;
+
+    public function test_it_return_all_choices()
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php:28]:
```suggestion
    /**
     * @test
     */
    public function it_return_all_choices(): void
```

File: src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+namespace Sylius\Bundle\AttributeBundle\Tests\Form\Type\AttributeType;
+
+use PHPUnit\Framework\Assert;
+use Prophecy\Prophecy\ProphecyInterface;
+use Sylius\Component\Resource\Translation\Provider\TranslationLocaleProviderInterface;
+use Symfony\Component\Form\ChoiceList\View\ChoiceView;
+use Symfony\Component\Form\PreloadedExtension;
+use Symfony\Component\Form\Test\TypeTestCase;
+
+class SelectAttributeTypeTest extends TypeTestCase
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php:23]:
```suggestion
final class SelectAttributeTypeTest extends TypeTestCase
```

File: src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php (hunk 5 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php:51]: Also, according to Symfony Coding Standards, we should place it as the first method in this file

File: src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php (hunk 6 above)
Comment (macintoshplus) [reply src/Sylius/Bundle/AttributeBundle/Tests/Form/Type/AttributeType/SelectAttributeTypeTest.php:43]: Fixed

**Comment by lchrusciel** [comment]
Hey Jean-Baptiste 👋

could you rebase your PR to the newest master? The build should be green then.

**Comment by macintoshplus** [comment]
> Hey Jean-Baptiste 
> 
> could you rebase your PR to the newest master? The build should be green then.

I need rebase to merge in 1.7 branch or master ? Actually the origin branch is 1.7 to merge into 1.7 branch.

**Comment by lchrusciel** [comment]
My mistake, of course, it should be rebased to 1.7.

**Comment by lchrusciel** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "fix/issue_11149"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Comment by macintoshplus** [comment]
@lchrusciel I have rebase my branch... The test error is not caused by this PR changes. How to relaunch to attempt to pass the test ?

**Comment by lchrusciel** [comment]
Tests at the bottom are rerun and passes. The problem is with static analysis: https://travis-ci.org/github/Sylius/Sylius/jobs/707744459#L2451

**Comment by macintoshplus** [comment]
I have fix the Psalm error (locally checked). I'm horrified by the need to add specific comment in code to pass static analysis.

**Comment by macintoshplus** [comment]
Enjoy! All check is done and no conflict! My PR can be merged @lchrusciel!

**Comment by lchrusciel** [comment]
Thanks, Jean-Baptiste! :tada:

**Pull request merged**